### PR TITLE
Add unit tests for Kademlia table.

### PR DESCRIPTION
**Pull request by kirkwood**
So far, these mostly exercise the distance function.

### CODE REVIEW

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      val nt = PeerTable(home)
+      nt.distance(home) should be (Some(8*nt.width))
+    }
+  }
+
+  val width = 32
+
+  // Make 8*width copies all of which differ in a single, distinct bit
+  def oneOffs(key: Seq[Byte]): Seq[Array[Byte]] = {
+    for (i <- 0 to width - 1; j <- 7 to 0 by -1) yield {
```
Comment (DoSOfRedRiver) [inline comm/src/test/scala/coop/rchain/kademlia/distance.scala]: We can replace `i <- 0 to width - 1` with `i <- 0 until width`, I think it's increases readability a little bit.

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      val k1 = new Array[Byte](key.size)
```
Comment (DoSOfRedRiver) [inline comm/src/test/scala/coop/rchain/kademlia/distance.scala]: I think we can replace `new` construction with `apply`, like: `val k1 = Array[Byte](key.size)`

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    s"#{PeerNode $skey}"
+  }
+}
+
+object b {
+  val rand = new scala.util.Random(System.currentTimeMillis)
+  def apply(i: Int): Byte = i.asInstanceOf[Byte]
+  def apply(s: String): Byte = b(Integer.parseInt(s, 2))
+  def rand(nbytes: Int): Seq[Byte] = {
+    val arr = new Array[Byte](nbytes)
```
Comment (DoSOfRedRiver) [inline comm/src/test/scala/coop/rchain/kademlia/distance.scala]: We can replace this withi `apply`, see below.

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+  override def toString = {
+    val skey = key.map("%02x" format _).mkString
+    s"#{PeerNode $skey}"
+  }
+}
+
+object b {
+  val rand = new scala.util.Random(System.currentTimeMillis)
+  def apply(i: Int): Byte = i.asInstanceOf[Byte]
```
Comment (DoSOfRedRiver) [inline comm/src/test/scala/coop/rchain/kademlia/distance.scala]: Instead `asInstanceOf` it is clearer to use `i.toByte` method, I think.

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      val k1 = new Array[Byte](key.size)
+      Array.copy(k1, 0, key.toArray, 0, key.size)
+      k1(i) = b(k1(i)^b(1 << j))
+      k1
+    }
+  }
+
+  def testKey(key: Array[Byte]): Boolean = {
+    val table = PeerTable(PeerNode(key))
+    oneOffs(key).map(table.distance(_)) == ((0 to (8*width - 1)).map(Option[Int](_)))
```
Comment (DoSOfRedRiver) [inline comm/src/test/scala/coop/rchain/kademlia/distance.scala]: Here `until` also can be used.

File: comm/src/test/scala/coop/rchain/kademlia/distance.scala (hunk 2 above)
Comment (kirkwood) [reply comm/src/test/scala/coop/rchain/kademlia/distance.scala]: Little different, I think:
```
scala> Array[Byte](3)
res0: Array[Byte] = Array(3)

scala> new Array[Byte](3)
res1: Array[Byte] = Array(0, 0, 0)
```
though one can do this
```
scala> Array.fill(3)(0.toByte)
res12: Array[Byte] = Array(0, 0, 0)
```
if you think it's better.

Comment (DoSOfRedRiver) [reply comm/src/test/scala/coop/rchain/kademlia/distance.scala]: Oh, my bad. Thanks for clarifying!

**Comment by kirkwood** [comment]
Successful run example:
```
[info] DistanceSpec:
[info] A PeerNode of width n byte
[info] - should have distance to itself equal to 8n
[info] A node with key all zeroes (0000000000000000000000000000000000000000000000000000000000000000)
[info] - should compute distance correctly
[info] A node with key all ones (ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
[info] - should compute distance correctly
[info] A node with random key (39d6f485b8a22b5758283247d499fafd434efd4d7887f632d186fe2b95de13bf)
[info] - should compute distance correctly
[info] An empty table
[info] - should have no peers
[info] - should return no values on lookup
[info] - should add any key at most once
[info] A table with peers at all distances
[info] - should have no empty slots
[info] - should return k peers on lookup
[info] - should not return sought peer on lookup
[info] Run completed in 277 milliseconds.
[info] Total number of tests run: 10
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 10, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 5 s, completed 11-Oct-2017 10:50:27 AM
```

**Pull request merged**